pallets/subtensor/src/lib.rs

```diff
@@ -917,15 +919,15 @@ pub mod pallet {
     pub type AdjustmentAlpha<T: Config> =
         StorageMap<_, Identity, u16, u64, ValueQuery, DefaultAdjustmentAlpha<T>>;
 
-    #[pallet::storage] // --- MAP (netuid, who) --> (hash, weight) | Returns the hash and weight committed by an account for a given netuid.
+    #[pallet::storage] // --- DMAP ( netuid, who ) --> BTreeMap<nonce, (hash, block_number)> | Stores weight commitments for each account in a subnet.
     pub type WeightCommits<T: Config> = StorageDoubleMap<
         _,
         Twox64Concat,
-        u16,
+        u16, // netuid
         Twox64Concat,
         T::AccountId,
-        (H256, u64),
-        OptionQuery,
+        BTreeMap<u64, (H256, u64)>, // nonce -> (hash, block_number)
+        ValueQuery,
     >;
```

Comment:
Does this need a migration?

Reply:
done. Can you please review the [implementation](https://github.com/opentensor/subtensor/pull/599/files#diff-c11128ae7d8317be661825c65161cc438085dba369eaeb4d77b58e36e2bab3fdR492) and [test ](https://github.com/opentensor/subtensor/pull/599/files#diff-2863c6b9602ad04e159f5db370384fd0f1a595f96777e57c0bc0dcd8fde53958R299)

Reply:
Looks good, you should also
- put the migration in the `OnRuntimeUpgrade` trait
- add the migration to the runtime
- set the in-code pallet storage version in the pallet (see https://github.com/paritytech/polkadot-sdk/blob/a466a490fb20945b834e6c1983cf37daba6fb3e4/substrate/frame/balances/src/lib.rs#L319)